security/pkg/nodeagent/caclient/credentials.go

```diff
@@ -16,17 +16,27 @@ package caclient
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"io/ioutil"
 	"strings"
 
 	"google.golang.org/grpc/credentials"
 
 	"istio.io/istio/pkg/security"
+	"istio.io/istio/security/pkg/stsservice"
+	"istio.io/istio/security/pkg/stsservice/server"
+	"istio.io/istio/security/pkg/stsservice/tokenmanager"
+	"istio.io/istio/security/pkg/stsservice/tokenmanager/google"
 	"istio.io/pkg/log"
 )
 
+const (
+	scope = "https://www.googleapis.com/auth/cloud-platform"
+)
+
 // TokenProvider is a grpc PerRPCCredentials that can be used to attach a JWT token to each gRPC call.
+// TokenProvider can be used for both XDS and CA connection, which may involve token exchange through STS.
```

Comment:
I don't think this is technically correct. Stackdriver ( or any envoy service using google gRPC ) will use STS , as implemented in istio-agent.

AFAIK the exchanges done for CA are not using STS - but vendor-specific protocol. If Google implements standard STS - we wouldn't need to have one in agent.

Reply:
The comments have been revised to reflect that CA does not use STS.